### PR TITLE
Add `remove` transform for hiding columns

### LINKED ISSUE #1867: Transformation for hiding columns
We want a DE transformation that selects a subset of columns.

### PR DESCRIPTION
**Pull request by dmos62**
Fixes #1867

The new transform's id is `remove` and it takes a list of aliases, in the same vein as the existing `select` transform.

Note, I moved a test function to a different file, and then added a test case to it. Sorry for the diff noise.

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `master` branch of the repository
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: db/transforms/base.py [hunk 1]
```diff
@@ -390,6 +390,48 @@ def _add_aliases_to_summarization_expr_field(
     return summarization
 
 
+class RemoveColumns(Transform):
+    """
+    We're implementing this transform in terms of selecting columns (since Postgres doesn't really
+    support "selecting everything except").
+    """
+    type = "remove"
```
Comment (kgodey) [inline db/transforms/base.py:398]: `remove` seems like a very generic name, can we call it `remove_columns` or something like that?

I'm not familiar with this area of the codebase so please ignore my comment if they are not helpful or conflict with the style.

Comment (dmos62) [reply db/transforms/base.py:398]: I considered that. As noted in the PR description, a close sibling of this transform is the `select` transform, which is just as generic, but happens to match an SQL term. That's the rationale behind calling this `remove`.

Standard SQL doesn't do "removing columns", though some variants have `SELECT * EXCEPT ...`, so another idea could be calling this transform `except`. We can also say `remove_columns`, but then let's rename `select` into `select_columns`.

Comment (kgodey) [reply db/transforms/base.py:398]: `hide` might be a good alternative (and also matches the UI). I think of `remove` as permanent somehow.

**Comment by dmos62** [comment]
Renamed `remove` to `hide`.

Converted to draft, because the non-master branch this is based on is not yet merged.

**Review by mathemancer — APPROVED** [review]
I think this one looks good. I note that this won't throw any validation error if one or more of the columns passed for "hiding" aren't there, but I think that's okay given our use case.

**Pull request merged**